src/index.js

```diff
@@ -30,50 +30,36 @@ export default function build (babel: Object): Object {
 
   return new Transformer("typecheck", {
     Program (node: Object, parent: Object, scope: Object) {
-      try {
-        this.traverse(visitors, {
-          constants: scope.getAllBindingsOfKind("const"),
-          subject: node,
-        });
-      }
-      catch (e) {
-        if (e instanceof SyntaxError) {
-          throw this.errorWithNode(e.message);
-        }
-        else {
-          throw e;
-        }
-      }
+      this.traverse(visitors, {
+        constants: scope.getAllBindingsOfKind("const"),
+        subject: node,
+      });
     },
     Function (node: Object, parent: Object, scope: Object) {
-      try {
-        if (node.typeParameters != null) {
-          throw this.errorWithNode('Type parameters are not supported');
-        }
+      if (node.typeParameters != null) {
```

Comment:
I don't think this should throw at all. We should not prevent the programmer from using constructs that flow supports but we don't (yet). We should just ignore syntax which we don't understand. In this case I think we should bail out of the function.